src/utils/checkName.ts

```diff
@@ -1,13 +1,17 @@
 const checkName = (name: string, nameList: string[]) => {
   let nameCheck = "passCheck";
   const checkStartEndSpace = /(^\s*)|(\s*$)/g;
+  // requirement for design name is only contain letter and number and less than 15 char
+  const validNamRegularExp = /^[A-Za-z0-9\s]{1,15}$/;

```

Comment:
Improvement hits:
The max length and even the rules can be passed by props into this helper util so that makes this util DRY without making other similar helpers.

Reply:
Thanks for the suggestion. The validation rules have been extracted to a validation function that passes to checkName. The validation rules can be configured using setValidation function.